### ISSUE TITLE
Markdown as default markup

### ISSUE THREAD

**Issue by villanella** [doorstop-dev/doorstop#592]
We want to edit Doorstop files as markdown with footer intstead of yml:
1) to use third-party editing tools. Some editors like VCS, Vim (vimwiki) support very useful markdown editing features:  tables formatting and alignment, custom syntax highlight etc;
2) to edit text without indents. Editing tables in markdown is very uncomfortable when we need to keep indents.

**Comment by jacebrowning** [comment]
This seems to be a duplicate of https://github.com/doorstop-dev/doorstop/issues/295.

**Comment by neerdoc** [comment]
@villanella Have you looked at #295? It is probably the best we can do since we do not want to go away from yaml as the file format.

**Comment by villanella** [comment]
I use this feature "markdown with yaml header" about 1 year in beta-b8 version. I can't upgrade Doorstop to newer versions, because markdown is supported only in beta releases, that aren't stable . Are you planning to support this feature in stable (not beta) release?

**Comment by neerdoc** [comment]
Always nice to hear from active users!

Yes, it will be supported in stable as it is already included in the development branch.

I've been way to busy with my day job to be able to spend enough time here for the last 2 years, but I've found some time now and is currently working on reaching the 3.0 milestone.

**Comment by robtaylor** [comment]
What I'm needing is a way to convert yml format items to markdown. My first attempts haven't gone well..
Any clues?

**Comment by robtaylor** [comment]
actually, this did the job:
```
import doorstop
root=doorstop.Document("./reqs", "./reqs")
for i in root.items:
     i.load()
     i.itemformat="markdown"
     i.save()

**Comment by robtaylor** [comment]
Ah, discovered my mistake: .doorstop.yml should contain:
```
settings:
    itemformat: markdown
```

**Comment by villanella** [comment]
> Ah, discovered my mistake: .doorstop.yml should contain:
> ```
> settings:
>     itemformat: markdown
> ```

Yes, I know these settings, but current release (not beta) version doesn't support markdown. 
And maybe it's not possible to set markdown as default markup for the whole project, without defining this parameter in each document's doorstop.yml.

**Comment by robtaylor** [comment]
> > Ah, discovered my mistake: .doorstop.yml should contain:
> > ```
> > settings:
> >     itemformat: markdown
> > ```
> 
> Yes, I know these settings, but current release (not beta) version doesn't support markdown. 

We use PDM for all our python projects, and so its easy to add the beta version as a dependency without messing up older projects (`pdm run doorstop`)

> And maybe it's not possible to set markdown as default markup for the whole project, without defining this parameter in each document's `.doorstop.yml`.

Yeah, I haven't figured out yet if it gets inherited down the tree... I *suspect* not, and I might throw a patch together if not.

So it sounds like really looking for is export and import from markdown? I can imagine that could get quite tricky. I'm thinking of having non-coders using the nice markdown editor in vscode to edit requirements. I'd be interested to hear your experiences!
